core/src/main/java/org/geysermc/geyser/session/GeyserSession.java

```diff
@@ -701,76 +694,20 @@ public void connect() {
     }
 
     public void authenticate(String username) {
-        authenticate(username, "");
-    }
-
-    public void authenticate(String username, String password) {
         if (loggedIn) {
             geyser.getLogger().severe(GeyserLocale.getLocaleStringLog("geyser.auth.already_loggedin", username));
             return;
         }
 
         loggingIn = true;
+        // Always replace spaces with underscores to avoid illegal nicknames, e.g. with GeyserConnect
+        protocol = new MinecraftProtocol(username.replace(' ', '_'));
```

Comment:
So we are replacing spaces regardless of authType now eh?

Reply:
I don't see a reason why not - mainly for e.g. GeyserConnect where you're basically forced to use either online or offline mode and no floodgate

Reply:
I know, I only mention it because of the precedence of #263

What you have here currently without the check is simpler code though, so I have no issue.